python/teams_ai/planner/planner.py

```diff
@@ -0,0 +1,40 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Any, Callable, Union
+from teams_ai.turn_state import TurnState
+
+class Planner(ABC):
+    """
+    interface implemented by all planners
+    """
+
+    @abstractclassmethod
+    async def generate_plan(self, turn_context:TurnContext, state: TurnState, prompt_name_or_template: Union[str, PromptTemplate], *, history_options: AIHistoryOptions) -> Plan:
+        """
+        Generates a plan based on the given turn state and prompt name or template.
+
+        Args:
+            turn_context (TurnContext): The turn context for current turn of conversation
+            state (TurnState): The current turn state.
+            prompt_name_or_template (Union[str, PromptTemplate]): The name of the prompt or a prompt template to use.
+            history_options (AIHistoryOptions): The options for the AI history.
+
+        Returns:
+            Plan: The generated plan.
+        """
+    
+
+    @abstractclassmethod
+    async def add_function(self, name: str, handler: Callable[[TurnContext, TurnState], Any], *, allow_overrides=False) -> None:
```

Comment:
is there a reason we have this here and not in the prompt manager class?

Reply:
The goal of this PR is update existing interface to move prompt management to planner. My understanding is there should be no prompt manager concept exposed to developers in this case (though we indeed have a prompt manage implementation). So I move the `add_function` functionality to planner since it's necessary when rendering the prompt template.